src/System.Runtime.Loader/tests/CollectibleAssemblyLoadContextTest.cs

```diff
@@ -0,0 +1,554 @@
+            Assert.Equal(2, exception.LoaderExceptions.Length);
+            Assert.True(exception.LoaderExceptions.Any(exp => exp.Message.Contains("Collectible type 'System.Runtime.Loader.Tests.TestClassNotSupported_ThreadStatic' may not have Thread or Context static members")));
+            Assert.True(exception.LoaderExceptions.Any(exp => exp.Message.Contains("Collectible type 'System.Runtime.Loader.Tests.TestClassNotSupported_FixedAddressValueType' has unsupported FixedAddressValueTypeAttribute applied to a field")));
+        }
+
+        class DelegateMarshallingTest : TestBase
+        {
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            public void Execute()
+            {
+                var exception = Assert.Throws<TargetInvocationException>(() =>
+                {
+                    var methodReference = _testClassTypes[0].GetMethod("TestDelegateMarshalling");
+                    Assert.NotNull(methodReference);
+                    methodReference.Invoke(null, null);
+                });
+                Assert.True(exception.InnerException is NotSupportedException);
+                Assert.True(exception.InnerException.Message.Contains("Delegate marshaling for types within collectible assemblies is not supported."));
+            }
+        }
+
+        [Fact]
+        public static void Unsupported_DelegateMarshalling()
+        {
+            var test = new DelegateMarshallingTest();
+            test.CreateContextAndLoadAssembly();
+            test.Execute();
+            test.UnloadAndClearContext();
+            test.CheckContextUnloaded();
+        }
+
+
+        class COMInteropTest : TestBase
+        {
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            public void Execute()
+            {
+                var exception = Assert.Throws<NotSupportedException>(() =>
+                {
+                    Type contextMenuType = _testClassTypes[0].GetNestedType("COMClass");
+                    Assert.NotNull(contextMenuType);
+                    Assert.True(contextMenuType.IsCOMObject);
+                    var contextMenu = Activator.CreateInstance(contextMenuType);
+                    Assert.NotNull(contextMenu);
+                });
+                Assert.True(exception.Message.Contains("COM Interop is not supported for collectible types."));
+            }
+        }
+
+        [Fact]
+        public static void Unsupported_COMInterop()
+        {
+            var test = new COMInteropTest();
+            test.CreateContextAndLoadAssembly();
+            test.Execute();
+            test.UnloadAndClearContext();
+            test.CheckContextUnloaded();
+        }
+
+        private class CollectibleChecker
+        {
+            private const int MaxGCRetry = 10;
+            private const int MaxWaitTimePerGCRetryInMilli = 10;
+
+            private readonly int _expectedCount;
+            private int _unloadCount;
```

Comment:
I do not understand what's the purpose of the `_unloadCount` field. We always recompute the precise `unloadCount` in GcAndCheck. Why do we need to remember it from the last time?